arch/cortex-m/src/lib.rs

```diff
@@ -128,24 +128,18 @@ pub unsafe extern "C" fn unhandled_interrupt() {
     panic!("Unhandled Interrupt. ISR {} is active.", interrupt_number);
 }
 
+/// Assembly function to initialize the .bss and .data sections in RAM.
+///
+/// We need to (unfortunately) do these operations in assembly because it is
+/// not valid to run Rust code without RAM initialized.
+///
+/// See <https://github.com/tock/tock/issues/2222> for more information.
 #[cfg(any(doc, all(target_arch = "arm", target_os = "none")))]
-extern "C" {
-    /// Assembly function to initialize the .bss and .data sections in RAM.
-    ///
-    /// We need to (unfortunately) do these operations in assembly because it is
-    /// not valid to run Rust code without RAM initialized.
-    ///
-    /// See <https://github.com/tock/tock/issues/2222> for more information.
-    pub fn initialize_ram_jump_to_main();
-}
-
-#[cfg(any(doc, all(target_arch = "arm", target_os = "none")))]
-core::arch::global_asm!(
-"
-    .section .initialize_ram_jump_to_main, \"ax\"
-    .global initialize_ram_jump_to_main
-    .thumb_func
-  initialize_ram_jump_to_main:
+#[unsafe(naked)]
+pub unsafe extern "C" fn initialize_ram_jump_to_main() {
+    use core::arch::naked_asm;
+    naked_asm!(
+        "
```

Comment:
This formatting rubs me the wrong way for a couple reasons:

1. Why is the quote on its own line?
2. It feels like the assembly should be indented one level further than it is. The quotation character being indented further than the assembly code is particularly weird.

Perhaps some tweaking can be done to make it less unusual? Granted, re-indenting will make this PR a LOT larger but I think it's the right thing to do.